cli/src/semgrep/engine.py

```diff
@@ -59,6 +63,12 @@ def decide_engine_type(
             ):
                 requested_engine = cls.PRO_INTRAFILE
 
+        if requested_engine != cls.OSS and supply_chain_only:
```

Comment:
I would leave a comment about how we do this because `PRO_INTERFILE` triggers `-j 1`